SubZero/src/model/state/CameraState.h

```diff
@@ -1,37 +1,75 @@
-/*
- * CameraState.h
- *
- *  Created on: Feb 7, 2015
- *      Author: carl
- */
-
-#ifndef CAMERASTATE_H_
-#define CAMERASTATE_H_
-#include "Observable.h"
-
-/**
- * CameraState is an Observable used by CameraModel to hold data sent from cameras.
- * It can notify any Observers when changes are made.
- */
-
-class CameraState : public Observable {
-private:
-	double myState;
-
-public:
-	/**
-	 * constructor
-	 */
-	CameraState();
-
-	/**
-	 * destructor
-	 */
-	virtual ~CameraState();
-
-	double getState();
-	void setState(myState);
-
-};
-
-#endif /* SRC_MODEL_STATE_CAMERASTATE_H_ */
+/*
+ * CameraState.h
+ *
+ *  Created on: Feb 7, 2015
+ *      Author: carl, james
+ */
+
+#ifndef CAMERASTATE_H_
+#define CAMERASTATE_H_
+#include "Observable.h"
+#include "State.h"
+#include "ImgData.h"	//rename it to whatever its called
+
+#include <list>
+#include <vector>
+
+
+/**
+ * CameraState is an Observable used by CameraModel to hold data sent from cameras.
+ */
+
+class CameraState : public State, public Observable {

```

Comment:
I think if State already inherits from Observable, then you don't need to make State's children be a child of Observable